cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/IndexTest.java

```diff
@@ -40,105 +40,92 @@ public void setUp() {
 
     @Test
     public void constructsIndexWithDefaultType() {
-        Index index = Index.getInstance(fieldNames, indexName);
+        Index index = new Index(fieldNames, indexName);
         assertThat(index.indexName, is("basic"));
         assertThat(index.fieldNames, is(Arrays.<FieldSort>asList(new FieldSort("name"), new FieldSort("age"))));
         assertThat(index.indexType, is(IndexType.JSON));
-        assertThat(index.indexSettings, is(nullValue()));
+        assertThat(index.tokenize, is(nullValue()));
     }
 
     @Test
     public void constructsIndexWithTextTypeDefaultSettings() {
-        Index index = Index.getInstance(fieldNames, indexName, IndexType.TEXT);
+        Index index = new Index(fieldNames, indexName, IndexType.TEXT);
         assertThat(index.indexName, is("basic"));
         assertThat(index.fieldNames, is(Arrays.<FieldSort>asList(new FieldSort("name"), new FieldSort("age"))));
         assertThat(index.indexType, is(IndexType.TEXT));
-        assertThat(index.indexSettings.size(), is(1));
-        assertThat(index.indexSettings.get("tokenize"), is("simple"));
+        assertThat(index.tokenize, is("simple"));
     }
 
-    @Test
+    @Test(expected = NullPointerException.class)
     public void returnsNullWhenNoFields() {
-        Index index = Index.getInstance(null, indexName);
+        Index index = new Index(null, indexName);
         assertThat(index, is(nullValue()));
 
-        index = Index.getInstance(new ArrayList<FieldSort>(), indexName);
+        index = new Index(new ArrayList<FieldSort>(), indexName);
         assertThat(index, is(nullValue()));
     }
 
     @Test
     public void returnsValueWhenIndexNameIsNull(){
-        Index index = Index.getInstance(fieldNames, null);
+        Index index = new Index(fieldNames, null);
         assertThat(index, is(notNullValue()));
     }
 
-    @Test
+    @Test(expected = IllegalArgumentException.class)
     public void returnsNullWhenNoIndexName() {
-        Index index = Index.getInstance(fieldNames, "");
-        assertThat(index, is(nullValue()));
-    }
-
-    @Test
-    public void returnsNullWhenInvalidIndexSettings() {
-        Map<String, String> indexSettings = new HashMap<String, String>();
-        indexSettings.put("foo", "bar");
-        Index index = Index.getInstance(fieldNames, indexName, IndexType.TEXT, indexSettings);
+        Index index = new Index(fieldNames, "");
         assertThat(index, is(nullValue()));
     }
 
     @Test
     public void correctlyIgnoresIndexSettings() {
-        Map<String, String> indexSettings = new HashMap<String, String>();
-        indexSettings.put("tokenize", "porter");
         // json indexes do not support index settings.  Index settings will be ignored.
-        Index index = Index.getInstance(fieldNames, indexName, IndexType.JSON, indexSettings);
-        assertThat(index.indexSettings, is(nullValue()));
+        Index index = new Index(fieldNames, indexName, IndexType.JSON, "porter");
```

Comment:
Looking at the API for this here, it looks a bit awkward, and odd that a constructor will not set a value rather than throw an `IllegalArgumentException`.